assets/libraries/common.rego

```diff
@@ -541,3 +541,49 @@ is_aws_ebs_optimized_by_default(instanceType) {
 	inArray(data.common_lib.aws_ebs_optimized_by_default, instanceType)
 }
 
+valid_for_IAM_engine_and_version_check(resource) {
```

Comment:
We can take advantage of this function and delete the similar function in cloudformation lib to something like:

```
valid_for_IAM_engine_and_version_check(resource, engine, engineVersion) {
	common_lib.valid_for_IAM_engine_and_costum_version_check(resource[engine], resource[engineVersion])
} else {
	engines_that_supports_iam := ["aurora", "aurora-mysql", "aurora-postgresql", "postgres", "mysql", "mariadb"]
	resource[engine] == engines_that_supports_iam[_]
	not common_lib.valid_key(resource, engineVersion)
}

example: valid_for_IAM_engine_and_version_check(resource, "engine", "engine_version")
```